axiom/cmd_appinfo_transmit.c

```diff
@@ -289,20 +289,18 @@ nr_status_t nr_cmd_appinfo_process_reply(const uint8_t* data,
       nrl_info(NRL_ACCT, "APPINFO reply disconnected app=" NRP_FMT,
                NRP_APPNAME(app->info.appname));
       return NR_SUCCESS;
-    //case APP_STATUS_INVALID_LICENSE:
-    //  app->state = NR_APP_INVALID;
-    //  nrl_error(NRL_ACCT,
-    //            "APPINFO reply invalid license app=" NRP_FMT
-    //            " please check your license "
-    //            "key and restart your web server.",
-    //            NRP_APPNAME(app->info.appname));
-    //  return NR_SUCCESS;
+    case APP_STATUS_INVALID_LICENSE:
+      app->state = NR_APP_INVALID;
+      nrl_error(NRL_ACCT,
+                "APPINFO reply invalid license app=" NRP_FMT
+                " please check your license "
+                "key and restart your web server.",
+                NRP_APPNAME(app->info.appname));
+      return NR_SUCCESS;
     case APP_STATUS_CONNECTED:
       /* Don't return here, instead break and continue below. */
       nrl_debug(NRL_ACCT, "APPINFO reply connected");
       break;
-    case APP_STATUS_RESTART:
```

Comment:
Shouldn't we handle this case?  Does the agent need to do something when the app is in a restart state?

Reply:
AppStateRestart is reported as AppStateUnkown. When the daemon enters AppStateRestart, [it very quickly switches over to AppStateUnkown during the restart](https://github.com/newrelic/newrelic-php-agent/pull/451/files/9796141f6fbfa6b4aabf1da58eb366e49b841bdc#diff-d34a767f316aaeded794ac3a231f867d3ffb42de19365ed338eace17f5876d94R677). While AppStateRestart is new in this PR, being in AppStateUnkown during restart is consistent with prior behavior.

Reply:
Ah!  Thanks for explaining :)